package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,355 @@
+Additionally corrections for finite size effects are sometimes employed along with varied means of estimating errors.
+The reader is directed to the following review, which describes many of the common pitfalls [Maginn2019]_. There are other ways to compute self diffusivity including from a Green-Kubo integral. At this point in time these methods are beyond the scope of this module
+
+
+References
+----------
+
+.. [Maginn2019] Maginn, E. J.; Messerly, R. A.; Carlson, D. J.; Roe, D. R.; Elliott, J. R. Best Practices for Computing Transport Properties 1. Self-Diffusivity and Viscosity from Equilibrium Molecular Dynamics [Article v1.0]. Living J. Comput. Mol. Sci. 2019, 1 (1).
+.. [Calandri2011] Calandrini, V.; Pellegrini, E.; Calligari, P.; Hinsen, K.; Kneller, G. R. NMoldyn-Interfacing Spectroscopic Experiments, Molecular Dynamics Simulations and Models for Time Correlation Functions. Collect. SFN 2011, 12, 201–232.
+.. [SO2015] https://stackoverflow.com/questions/34222272/computing-mean-square-displacement-using-python-and-fft
+
+Classes and Functions
+---------------------
+
+.. autoclass:: MeanSquaredDisplacement
+    :members:
+
+"""
+
+from __future__ import division, absolute_import
+from six.moves import zip
+
+import os
+import errno
+import warnings
+import bz2
+import functools
+
+import numpy as np
+from numpy.fft import fft,ifft
+import logging
+import MDAnalysis
+
+class MeanSquaredDisplacement(object):
+    r"""Class representing Mean Squared Displacement
+
+    Parameters
+    ----------
+    u : Universe
+        An MDAnalysis :class:`Universe`
+    selection : str
+        An MDAnalysis selection string
+    msd_type : str
+        The dimensions to use for the msd calculation:
+        one of ('xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z')
+    fft : bool
+        Use a fast FFT based algorithm for computation of the MSD
+
+    Returns
+    -------
+    MSD : :class:`MeanSquaredDisplacement`
+        the MSD class
+    
+    """
+    # Attributes
+    # ----------
+    # _dim_fac : int
+    #     dimensionality of the MSD
+    # _dim : arraylike
+    #     array used to slice the trajectory along the xyz axis to acheive the correct dimensionality
+    # _position_array : :class:`np.ndarray`
+    #     positions used to calculate the MSD
+    # N_particles : int
+    #     number of particles 
+
+
+    def __init__(self, u, selection, msd_type='xyz', fft=True):
+
+        #args
+        self.u = u
+        self.selection = selection
+        self.msd_type = msd_type
+        self.fft = fft
+        
+        #local
+        self.dim_fac = 0
+        self._dim = None
+        self.atoms = None
+        self.n_frames = len(self.u.trajectory)
+        self._position_array = None
+
+        #result
+        self.timeseries = None
+        #prep
+        self._prepare()
+
+            
+    def _prepare(self):
+        self.parse_msd_type()
+        self.select_reference_positions()
+        
+    def parse_msd_type(self):
```

Comment:
anything a user isn't expected to use should get prefixed by an underscore or they'll wonder how to use it